src/ferui-components/tree-view/interfaces.ts

```diff
@@ -0,0 +1,108 @@
+import { TemplateRef } from '@angular/core';
+
+/**
+ * Tree Node Data Interface
+ */
+export interface TreeNodeData<T> {
+  data: T;
+  label: string;
+}
+
+/**
+ * Paging Params Interface
+ */
+export interface PagingParams {
+  offset: number;
+  limit: number;
+}
+
+/**
+ * Tree Node Data Retriever Interface for client-side tree view
+ */
+export interface TreeNodeDataRetriever<T> {
+  // Get promise that returns TreeNodeData children array from Tree Node
+  getChildNodeData(parent: TreeNode<T>): Promise<Array<TreeNodeData<T>>>;
+  // Get promise that returns whether TreeNode has children nodes
+  hasChildNodes(parent: TreeNode<T>): Promise<boolean>;
+  // based on the current TreeNode and its status, we allow the developer to give us an icon template
+  getIconTemplate?(treeNode: TreeNode<T>): TemplateRef<any>;
+  // If developer wishes to give us a template ref we shall render this template in the view
+  getNodeTemplate?(): TemplateRef<any>;
+}
+
+/**
+ * Paged Tree Node Data Retriever for server-side tree view
+ */
+export interface PagedTreeNodeDataRetriever<T> extends TreeNodeDataRetriever<T> {
+  getPagedChildNodeData(parent: TreeNode<T>, pagingParams: PagingParams): Promise<Array<TreeNodeData<T>>>;
+}
+
+/**
+ * Tree Node Interface
+ */
+export interface TreeNode<T> {
+  data: TreeNodeData<T>;
+  selected: boolean;
+  expanded: boolean;
+  children: Array<TreeNode<T>>;
+  allChildrenLoaded: boolean;
+  parent: TreeNode<T> | null;
+  showLoader: boolean;
+  loadError: boolean;
+}
+
+/**
+ * Non root tree node empty instance
+ */
+export class NonRootTreeNode<T> implements TreeNodeData<T> {
+  data: T;
+  label: string;
+}
+
+/**
+ * Tree View Event Interface
+ */
+export interface TreeViewEvent<T> {
+  getNode(): TreeNode<T>;
+  getType(): TreeViewEventType;
+}
+
+/**
+ * Tree View Event Type enum
+ */
+export enum TreeViewEventType {
+  NODE_CLICKED,
+  NODE_EXPANDED,
+  NODE_COLLAPSED,
+}
+
+/**
+ * Tree View Configuration Interface
+ */
+export interface TreeViewConfiguration {
+  width?: string;
```

Comment:
Why do we have the width and height there if it's already part of FuiTreeViewComponentStyles ?

Reply:
I expected that the custom template might be defined in this config vs. with the data retriever, but maybe it makes sense there too :thinking:

Reply:
Thats an internal interface just so i could have a type for the styles. I moved it to be internal now

Reply:
@ariquinones if you have internal only interfaces, please do not add them to the `index.ts` file so that they're not exported publicly.

Reply:
Yes, these internal interfaces aren't being exported to the public api